src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -53,6 +53,9 @@ def from_native(self, data):
                 'Ensure at least one action is enabled and all required fields are filled in.'
             )
 
+        # Update data from cleaned form values
+        data.update(form.cleaned_data)
```

Comment:
Data needs to be updated with the cleaned value from the form since we sanitize the channel / username during cleaning.